apps/logging/serializers.py

```diff
@@ -33,3 +33,74 @@ def __str__(self):
             }
         }
         return json.dumps(result)
+
+
+class Request:
+    # requests.PrepairedRequest
+    req = None
+
+    def __init__(self, request):
+        self.req = request
+
+    def uuid(self):
+        return self.req.headers.get('BlueButton-OriginalQueryId')
+
+    def user(self):
+        return self.req.headers.get('BlueButton-BeneficiaryId')
+
+    def start_time(self):
+        return self.req.headers.get('BlueButton-OriginalQueryTimestamp')
+
+    def application(self):
+        return {
+            "name": self.req.headers.get('BlueButton-Application'),
+            "id": self.req.headers.get('BlueButton-ApplicationId'),
+            "user": {
+                "id": self.req.headers.get('BlueButton-DeveloperId'),
+            },
+        }
+
+    def path(self):
+        return self.req.headers.get('BlueButton-OriginalUrl')
+
+    def to_dict(self):
+        result = {
+            "uuid": self.uuid(),
+            "user": self.user(),
+            "start_time": self.start_time(),
+            "application": self.application(),
+            "path": self.path(),
+        }
+        return result
+
+    def __str__(self):
+        return json.dumps(self.to_dict())
+
+
+class Response:
+    resp = None
+    req = None
+
+    def __init__(self, response, request=None):
+        self.resp = response
+        self.req = request
+
+    def code(self):
+        return self.resp.status_code
+
+    def size(self):
+        return len(self.resp.content)
+
+    def elapsed(self):
+        return self.resp.elapsed.total_seconds()
+
+    def to_dict(self):
+        return {
+            "code": self.code(),
+            "size": self.size(),
+            "elapsed": self.elapsed(),
+        }
+
+    def __str__(self):
+        result = {**self.to_dict(), **self.req.to_dict()}
```

Comment:
This will fail if `Response` is initialized without providing a `request`. Worth making `request` (in `__init__` above) a required arg if we always expect a `Response` to have corresponding `Request`?

Reply:
good catch! updated to create and use a default empty object.